packages/ckeditor5-paste-from-office/src/filters/table.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module paste-from-office/filters/parse
+ */
+import type { UpcastWriter, ViewDocumentFragment } from 'ckeditor5/src/engine';
+
+/**
+ * Set alignment for table pasted from MS Word.
+ *
+ * @param documentFragment The view structure to be transformed.
+ */
+export function tableAlignmentFilter( documentFragment: ViewDocumentFragment, writer: UpcastWriter ): void {
+	for ( const item of documentFragment.getChildren() ) {
```

Comment:
With the shallow iterator like this, the tables nested deeper than in one container will not be caught. I don't have an Office license now, but I imagine things like table inside the blockquote or just a table inside a table are possible. So we need to make this look as deep as possible.

You can get inspired by how it's done in other filters: https://github.com/ckeditor/ckeditor5/blob/cdebba2da7fe5746833dcd3e51e72115bb99c5f6/packages/ckeditor5-paste-from-office/src/filters/image.ts#L67-L75

Reply:
There is no need indeed. MS Word doesn't allow you to have table in other blocks as list for example. Blockquote is not a block in the MS Word, it just spacing between corner of document and your paragraph or other elements. How to add blockquote in MS Word could check it here https://www.howtogeek.com/697529/how-to-add-block-quotes-in-microsoft-word/

Reply:
> MS Word doesn't allow you to have table in other blocks

I confirmed it with the converters team, so the only scenario where you can have a table inside a block element is a table inside a table. I agree it would be a rare scenario, so we can go with the performance micro-optimisation you've proposed. I'll extract an issue for the mentioned case so we stay aware of it.

Reply:
Reported in #14609.